src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java

```diff
@@ -403,28 +383,16 @@ private Properties getBuildFilterProperties()
      * exists, it will be copied to the {@code META-INF} directory and
      * renamed accordingly.
      *
-     * @param resource         the resource to copy
+     * @param resources        the resources to copy
      * @param webappDirectory  the target directory
-     * @throws java.io.IOException if an error occurred while copying webResources
-     */
-    public void copyResources(Resource resource, File webappDirectory, Properties filterProperties)
-        throws IOException {
-        if (!resource.getDirectory().equals(webappDirectory.getPath())) {
-            getLog().info("Copy webapp webResources to " + webappDirectory.getAbsolutePath());
-            if (webappDirectory.exists()) {
-                String[] fileNames = getWarFiles(resource);
-                for (String fileName : fileNames) {
-                    if (resource.isFiltering()) {
-                        copyFilteredFile(new File(resource.getDirectory(), fileName),
-                            new File(webappDirectory, fileName), null, getFilterWrappers(),
-                            filterProperties);
-                    } else {
-                        FileUtils.copyFileIfModified(new File(resource.getDirectory(), fileName),
-                            new File(webappDirectory, fileName));
-                    }
-                }
-            }
-        }
+     * @throws MavenFilteringException if an error occurred while copying webResources
+     */
+    public void copyResourcesWithFiltering(List<Resource> resources, File webappDirectory) throws MavenFilteringException {
+        MavenResourcesExecution mavenResourcesExecution = 
+                new MavenResourcesExecution(resources, webappDirectory, project, StandardCharsets.UTF_8.name(), filters,
+                                              Collections.emptyList(), session);
+        mavenResourcesExecution.setEscapeString("\\");
```

Comment:
this means we can escape things - which is what you would get with regular maven filtering.

technically this is a breaking change - but looking through projects of the 5 I found adding `webResources` non used filtering anyway... so making this compatible would appear to be overkill

An argument could be made therefor to remove this code entirely - but as the `webResources` are of type [`Resource`](https://maven.apache.org/ref/3.9.0/maven-model/apidocs/org/apache/maven/model/Resource.html) it would mean we would take options that would be discarded.